src/python/pathops/_pathops.pyx

```diff
@@ -887,20 +899,24 @@ cdef class PathPen:
 
     cpdef moveTo(self, pt):
         self.path.moveTo(pt[0], pt[1])
+        self.path.originalOnCurvePoints.add(pt)
 
     cpdef lineTo(self, pt):
         self.path.lineTo(pt[0], pt[1])
+        self.path.originalOnCurvePoints.add(pt)
 
     cpdef curveTo(self, pt1, pt2, pt3):
         # support BasePen "super-beziers"? Nah.
         self.path.cubicTo(
             pt1[0], pt1[1],
             pt2[0], pt2[1],
             pt3[0], pt3[1])
+        self.path.originalOnCurvePoints.add(pt3)
 
     def qCurveTo(self, *points):
         for pt1, pt2 in _decompose_quadratic_segment(points):
             self._qCurveToOne(pt1, pt2)
+        self.path.originalOnCurvePoints.add(points[-1])
```

Comment:
I realize now that I could probably only keep track of the original on-curves at the end of qCurves, instead of all types of original on-curves?

Reply:
yeah, if we have to remember oncurve points, it's probably better to only do it for when we actually need it, i.e. for qCurve segments only where they can be "implied", the other segment types in pen protocol are all explicit